docs/design/task-minAvailable-dependson-design.md

```diff
@@ -0,0 +1,341 @@
+# Task minAvailable and dependsOn design
+
+solution [issue 2895](https://github.com/volcano-sh/volcano/issues/2895) and
+[issue 2921](https://github.com/volcano-sh/volcano/issues/2921) and 
+[issue 2944](https://github.com/volcano-sh/volcano/issues/2944)
+
+# Abstract
+This document mainly focuses on the detailed design of the two features supported by the volcano, 
+one is task minAvailable; the other is dependsOn between tasks.
+
+- Check vcjob via webhook, job MinAvailable and task MinAvailable are related as follows：
+
+|                      | task.MinAvailable == 0                                                                                                                      | task.MinAvailable != 0                                             |
+|----------------------|---------------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------|
+| job.MinAvailable ==0 | job.MinAvailable = sum(task.Replacas) (webhook mutate)                                                                                      | job.MinAvailable = sum(task.MinAvailable) (webhook mutate)         |
+| job.MinAvailable !=0 | one(task) task.MinAvailable=job.MinAvailable (webhook mutate)  <br/>multiple(task) job.MinAvailable = sum(task.Replacas) (webhook validate) | job.MinAvailable = sum(task.MinAvailable)       (webhook validate) |
+
+- When scheduling dependsOn job to execute gang check or Valid, we need to skip the checks that the task have not yet created pods
+  
+The modification involving the function is as follows：
+
+<img src="images/minAvailable-dependsOn.png" alt="当前方案1" width="1100"><br>
+
+Overall, when a user submits a vcjob, it will go through the following process:
+
+- Volcano webhook execute validate and mutate, find the minimum number of tasks;
+  webhook mutate checks vcjob dependsOn, if it depends on job, set annotation `dependent-job=true`.
+- The controller of the volcano will count the poggroup according to the `tasks.minAvailable` and `tasks.resources` information.
+  When the controller watch poggroup reaches the Inqueue state, all pods are created; when the number of running pod
+  reaches the number of `job.minAvailables`, the state of the job is set to Running. 
+- Volcano scheduling batch pods, scheduler will check the number of pods and sorting pods; when scheduling pods, 
+  it will be scheduled according to the sorted pod `one by one`, when the minimum number requirements are met,
+  the scheduling is considered successful, to bind. 
+- Skip the dependent job gang check that have not yet created pods.
+
+# user-guide
+
+When the user uses the vcjob minAvailable feature, only should set each `task.minAvailable` value, if not set `task.minAvailable`,
+the default `task.minAvailable=task.replaces`.
```

Comment:
task.minavailable is not equal to task.replaces if job.minavailable is set

Reply:
There is only one task in job, job.minavailable or task.minavailable set any one is ok, but  Multitasking is ambiguous.